cmake/onnxruntime_unittests.cmake

```diff
@@ -403,7 +403,7 @@ endif()
 set(all_dependencies ${onnxruntime_test_providers_dependencies} )
 
   if (onnxruntime_ENABLE_TRAINING)
-    if(${CMAKE_BUILD_TYPE} MATCHES "Debug" AND NOT onnxruntime_USE_CUDA)
+    if(${CMAKE_BUILD_TYPE} MATCHES "Debug")
```

Comment:
NO. Don't use CMAKE_BUILD_TYPE in any cmake files as possible as you can.

Reply:
See: https://cmake.org/cmake/help/v3.17/prop_gbl/GENERATOR_IS_MULTI_CONFIG.html

It doesn't work well with Visual Studio.